astropy/coordinates/builtin_frames/galactocentric.py

```diff
@@ -82,100 +102,231 @@ class galactocentric_frame_defaults(ScienceState):
+                    ),
+                    "galcen_distance": 8.122 * u.kpc,
+                    "galcen_v_sun": r.CartesianDifferential(
+                        [12.9, 245.6, 7.78] * (u.km / u.s)
+                    ),
+                    "z_sun": 20.8 * u.pc,
+                    "roll": 0 * u.deg,
+                }
+            ),
+            "references": MappingProxyType(
+                {
+                    "galcen_coord": "http://adsabs.harvard.edu/abs/2004ApJ...616..872R",
+                    "galcen_distance": "https://ui.adsabs.harvard.edu/abs/2018A%26A...615L..15G",
+                    "galcen_v_sun": [
+                        "https://ui.adsabs.harvard.edu/abs/2018RNAAS...2..210D",
+                        "https://ui.adsabs.harvard.edu/abs/2018A%26A...615L..15G",
+                        "https://ui.adsabs.harvard.edu/abs/2004ApJ...616..872R",
+                    ],
+                    "z_sun": "https://ui.adsabs.harvard.edu/abs/2019MNRAS.482.1417B",
+                    "roll": None,
+                }
+            ),
+        },
+        "pre-v4.0": {
+            "parameters": MappingProxyType(
+                {
+                    "galcen_coord": ICRS(
+                        ra=266.4051 * u.degree, dec=-28.936175 * u.degree
+                    ),
+                    "galcen_distance": 8.3 * u.kpc,
+                    "galcen_v_sun": r.CartesianDifferential(
+                        [11.1, 220 + 12.24, 7.25] * (u.km / u.s)
+                    ),
+                    "z_sun": 27.0 * u.pc,
+                    "roll": 0 * u.deg,
+                }
+            ),
+            "references": MappingProxyType(
+                {
+                    "galcen_coord": "http://adsabs.harvard.edu/abs/2004ApJ...616..872R",
+                    "galcen_distance": "https://ui.adsabs.harvard.edu/#abs/2009ApJ...692.1075G",
+                    "galcen_v_sun": [
+                        "https://ui.adsabs.harvard.edu/#abs/2010MNRAS.403.1829S",
+                        "https://ui.adsabs.harvard.edu/#abs/2015ApJS..216...29B",
+                    ],
+                    "z_sun": "https://ui.adsabs.harvard.edu/#abs/2001ApJ...553..184C",
+                    "roll": None,
+                }
+            ),
+        },
+    }
+
+    @classproperty  # read-only
+    def parameters(cls):
+        return cls._value
+
+    @classproperty  # read-only
+    def references(cls):
+        return cls._references
 
     @classmethod
-    def get_solar_params_from_string(cls, arg):
-        """Return Galactocentric solar parameters given string names for the
-        parameter sets.
+    def get_from_registry(cls, name: str):
         """
+        Return Galactocentric solar parameters and metadata given string names
+        for the parameter sets. This method ensures the returned state is
+        mutable, but the registry state remains unchanged. This is the
+        preferred means of accessing the registry.
```

Comment:
I'd delete "This is the preferred..." - you have it on top.

Reply:
✔︎